frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -84,27 +84,42 @@ const getNextUntitledNumber = (tabs: QueryTab[]): number => {
     }
     return untitledNumbers.length + 1
 }
-
 const getStorageItem = async (key: string): Promise<string | null> => {
-    const dbValue = await get(key)
-
-    if (dbValue) {
-        return dbValue
-    }
-
+    // 1. Try localStorage first (for backward compatibility)
     const lsValue = localStorage.getItem(key)
-
     if (lsValue) {
-        await set(key, lsValue)
-        localStorage.removeItem(key)
+        // Migrate to IndexedDB and clean up localStorage
+        try {
+            await set(key, lsValue)
+            localStorage.removeItem(key)
+        } catch (error) {
+            console.warn('Failed to migrate localStorage to IndexedDB:', error)
+        }
         return lsValue
     }
-
+    // 2. Try IndexedDB (primary storage)
+    try {
+        const dbValue = await get(key)
+        if (dbValue) {
+            return dbValue
+        }
+    } catch (error) {
+        console.warn('Failed to read from IndexedDB:', error)
```

Comment:
These warns don't really do much - it would be better to track these in `posthog.capture_exception(...)` instead

Reply:
Also, some more context on the error message would be helpful